modules/lvm2/udiskslvm2moduleiface.c

```diff
@@ -154,7 +154,6 @@ lvm_update_from_variant (GPid      pid,
           udisks_linux_volume_group_object_destroy (group);
           g_dbus_object_manager_server_unexport (manager,
                                                  g_dbus_object_get_object_path (G_DBUS_OBJECT (group)));
-          g_object_unref (G_OBJECT (group));
```

Comment:
Good question!

Looking at the history, commit c179109d4e33186fb323d5dcd5e81eed785296ec added this extra `g_object_unref` to fix a memory leak that my commit is backing out, commit c179109d4e33186fb323d5dcd5e81eed785296ec also added the one in this code block which is virtually identical:

``` C
while (g_hash_table_iter_next (&volume_iter, &key, &value))
    {
      const gchar *name = key;
      UDisksLinuxLogicalVolumeObject *volume = value;

      if (!g_hash_table_contains (new_lvs, name))
        {
          g_dbus_object_manager_server_unexport (manager,
                                                 g_dbus_object_get_object_path (G_DBUS_OBJECT (volume)));
          g_hash_table_iter_remove (&volume_iter);

          g_object_unref (G_OBJECT (volume));
        }
    }
```
The one I'm backing out with my commit is causing an `GLib-GObject-CRITICAL` because it's already been destroyed.  However, I'm not sure why we aren't getting a similar condition with this other code path except that the reference counts must be >= 2 when we go through the code path.

At the moment, the only thing I am sure about is that we don't have a good handle on managing the lifetimes of the objects.  It seems more akin to sprinkling gets and puts until nothing blows up and nothing appears to leak.  I don't feel very confident about the validity of this change anymore.